### PR TITLE
Initial attempt at Github Actions

### PR DESCRIPTION
**Pull request by bdunne**
Part of: https://github.com/ManageIQ/manageiq/issues/21620

I'll open a follow-up to remove the Travis config

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  build:
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up environment
+      run: |
+        echo "TRAVIS_PULL_REQUEST=${{ github.base_ref == '' && 'false' || 'true' }}" >> $GITHUB_ENV
+        echo "REGISTRY_PASSWORD=${{ secrets.REGISTRY_PASSWORD }}" >> $GITHUB_ENV
+        echo "REGISTRY_USERNAME=${{ secrets.REGISTRY_USERNAME }}" >> $GITHUB_ENV
+    - name: Run container build script
+      run: bin/build_container_image
```
Comment (jrafanie) [inline .github/workflows/ci.yaml:31]: Is this doing a build on pull requests in addition to pushes?

Comment (bdunne) [reply .github/workflows/ci.yaml:31]: Yes, it builds on every PR, but only pushes to the registry on merges

Comment (bdunne) [reply .github/workflows/ci.yaml:34]: I split it so that the credentials shouldn't even show up on PR run.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Run tests
+      run: bundle exec rake
+  build:
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up environment
+      run: |
+        echo "TRAVIS_PULL_REQUEST=${{ github.base_ref == '' && 'false' || 'true' }}" >> $GITHUB_ENV
+    - name: Set up registry credentials
+      if: ${{ github.base_ref == '' }}
```
Comment (jrafanie) [inline .github/workflows/ci.yaml:29]: I'm guessing this is the "if not a PR" logic, right?

Comment (bdunne) [reply .github/workflows/ci.yaml:29]: Yes, that's the closest thing I could find for it.

**Review by jrafanie — APPROVED** [review]
LGTM.  :shipit:

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 64c31c3d6854bab2d1abdc5678e8f1b7f42b00e0.

```text
commit 64c31c3d6854bab2d1abdc5678e8f1b7f42b00e0
Author: Joe Rafaniello <jrafanie@users.noreply.github.com>
Date:   Fri Dec 17 16:18:22 2021 -0500

    Merge pull request #235 from bdunne/initial_github_actions
    
    (cherry picked from commit f7ad87c566f0f7c64e3ff678b6fb7c4773704f5d)
```

**Pull request merged**